### PR TITLE
Add package liblouis version 3.24.0

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Add package liblouis version 3.22.0

###### Change Log  

- NA

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA
###### Test Methodology

Buddy Build
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=290720&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=290719&view=results)

### CODE REVIEW

**Comment by suresh-thelkar** [comment]
Please note that all linter warnings are fixed except a few false warnings.

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 1]
```diff
@@ -0,0 +1,407 @@
+# Turn off the brp-python-bytecompile script
+%global __os_install_post %(echo '%{__os_install_post}' | sed -e 's!/usr/lib[^[:space:]]*/brp-python-bytecompile[[:space:]].*$!!g')
+Summary:        Braille translation and back-translation library
+Name:           liblouis
+Version:        3.22.0
+Release:        2%{?dist}
+License:        LGPLv3+
```
Comment (oliviacrain) [inline SPECS-EXTENDED/liblouis/liblouis.spec:7]: Small nit: We're trying to slowly change these license tags to use [SPDX expressions](https://spdx.org/licenses/). Mind using the SPDX name for this and other licenses?

```suggestion
License:        LGPL-3.0-or-later
```

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+Liblouis has features to support screen-reading programs. This has led to its
+use in two open-source screen readers, NVDA and Orca. It is also used in some
+commercial assistive technology applications for example by ViewPlus.
+
+Liblouis is based on the translation routines in the BRLTTY screen reader for
+Linux. It has, however, gone far beyond these routines.
+
+%package        devel
+Summary:        Development files for %{name}
+License:        LGPLv3+
```
Comment (oliviacrain) [inline SPECS-EXTENDED/liblouis/liblouis.spec:44]:
```suggestion
License:        LGPL-3.0-or-later
```

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+Requires:       %{name}%{?_isa} = %{version}-%{release}
+Requires:       pkgconfig
+
+%description    devel
+The %{name}-devel package contains libraries and header files for
+developing applications that use %{name}.
+
+%package        utils
+Summary:        Command-line utilities to test %{name}
+License:        GPLv3+
```
Comment (oliviacrain) [inline SPECS-EXTENDED/liblouis/liblouis.spec:54]:
```suggestion
License:        GPL-3.0-or-later
```

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+
+%description    utils
+Six test programs are provided as part of the liblouis package. They
+are intended for testing liblouis and for debugging tables. None of
+them is suitable for braille transcription.
+
+%package -n python3-louis
+%{?python_provide:%python_provide python3-louis}
+Summary:        Python 3 language bindings for %{name}
+License:        LGPLv3+
```
Comment (oliviacrain) [inline SPECS-EXTENDED/liblouis/liblouis.spec:65]:
```suggestion
License:        LGPL-3.0-or-later
```

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 5]
```diff
@@ -0,0 +1,407 @@
+# Turn off the brp-python-bytecompile script
+%global __os_install_post %(echo '%{__os_install_post}' | sed -e 's!/usr/lib[^[:space:]]*/brp-python-bytecompile[[:space:]].*$!!g')
+Summary:        Braille translation and back-translation library
+Name:           liblouis
+Version:        3.22.0
+Release:        2%{?dist}
+License:        LGPLv3+
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            http://liblouis.org
```
Comment (oliviacrain) [inline SPECS-EXTENDED/liblouis/liblouis.spec:10]: Can we change this to point somewhere else? Judging by the registration history of the URL, they're in the middle of moving domain registrars and this currently leads to an ad-infested registrar page.

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+License:        LGPLv3+
+Requires:       %{name} = %{version}-%{release}
+BuildArch:      noarch
+
+%description doc
+This package provides the documentation for liblouis.
+%endif
+
+%global __os_install_post %{nil}
+%global debug_package %{nil}
```
Comment (oliviacrain) [inline SPECS-EXTENDED/liblouis/liblouis.spec:86]: Can we document the rationale for these flags?

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+rm -rf %{buildroot}/%{_defaultdocdir}/%{name}/
+
+# Replace table files with identical content by symlinks
+%fdupes -s %{buildroot}%{_datadir}/%{name}/tables/
+
+cd python/louis
+
+install -d %{buildroot}%{python3_sitelib}/louis
+install -pm 0644 __init__.py %{buildroot}%{python3_sitelib}/louis/
+%py_byte_compile %{__python3} %{buildroot}%{python3_sitelib}/louis/
```
Comment (oliviacrain) [inline SPECS-EXTENDED/liblouis/liblouis.spec:122]: I'm not sure I understand what's happening here- why are we disabling automatic compilation as the beginning of the spec, and then doing it manually here?

File: SPECS-EXTENDED/liblouis/liblouis.spec (hunk 1 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/liblouis/liblouis.spec]: Thanks for suggesting this. I have modified the license tags as per SPDX expressions

File: SPECS-EXTENDED/liblouis/liblouis.spec (hunk 5 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/liblouis/liblouis.spec]: I have modified the URL to the following. Hope that is fine. 
https://github.com/liblouis/liblouis

File: SPECS-EXTENDED/liblouis/liblouis.spec [hunk 8]
```diff
@@ -0,0 +1,407 @@
+# Turn off the brp-python-bytecompile script
+%global __os_install_post %(echo '%{__os_install_post}' | sed -e 's!/usr/lib[^[:space:]]*/brp-python-bytecompile[[:space:]].*$!!g')
```
[the remark this answers is not in the document]
Comment (suresh-thelkar) [reply SPECS-EXTENDED/liblouis/liblouis.spec]: Yes you are right. In fact there is no need to disable automatic byte compilation and hence I removed the above portion of the code. Please look at the latest commit.

File: SPECS-EXTENDED/liblouis/liblouis.spec (hunk 6 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/liblouis/liblouis.spec]: I realized that these two macros are not needed and removed them from the latest commit. Please take a look.

File: SPECS-EXTENDED/liblouis/liblouis.spec (hunk 7 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/liblouis/liblouis.spec]: I have analyzed further on this. It seems the following guideline is used by the people from where we imported our spec file initially for the various reasons.

"One MUST disable their automatic compilation, and compile them explicitly by the %py_byte_compile macro if and only if the bytecompilation is desired."

But when it comes to Mariner I believe we no need to follow the above guideline and hence I removed the following code. 
 `%py_byte_compile %{__python3} %{buildroot}%{python3_sitelib}/louis/`

After successful build, I have observed that In our case the respective *.py files are automatically byte compiled which is what we wanted.  
`rpm -qlp RPMS/noarch/python3-louis-3.24.0-2.cm2.noarch.rpm
/usr/lib/python3.9/site-packages/louis
/usr/lib/python3.9/site-packages/louis/__init__.py
/usr/lib/python3.9/site-packages/louis/__pycache__
/usr/lib/python3.9/site-packages/louis/__pycache__/__init__.cpython-39.opt-1.pyc
/usr/lib/python3.9/site-packages/louis/__pycache__/__init__.cpython-39.pyc
`

**Comment by suresh-thelkar** [comment]
@oliviacrain, Thank for the review. I have addressed all your comments. Please take a look at the latest commit/s and let me know if you have any further comments. 

I have also run Buddy builds after implementing the review comments. Both of them passed. 
[AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=291997&view=results)
[ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=291998&view=results)

**Comment by suresh-thelkar** [comment]
I have once again ran Buddy Builds after fixing linter warnings and all the commits. Both the following builds are passing. 
[AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=292006&view=results)
[ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=292007&view=results)

**Review by oliviacrain — APPROVED** [review]
Looks great!

**Comment by suresh-thelkar** [comment]
> Looks great!

Thank you for the approval.

**Comment by suresh-thelkar** [comment]
I have once again ran buddy builds after rebasing. Here are the results. Both of them passed.
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=293966&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=293954&view=results)

**Pull request merged**